packages/remix-dev/vite/plugin.ts

```diff
@@ -504,7 +506,9 @@ export const reactRouterVitePlugin: ReactRouterVitePlugin = (_config) => {
       )};
       export const basename = ${JSON.stringify(ctx.reactRouterConfig.basename)};
       export const future = ${JSON.stringify(ctx.reactRouterConfig.future)};
-      export const isSpaMode = ${!ctx.reactRouterConfig.ssr};
+      export const isSpaMode = ${
+        !ctx.reactRouterConfig.ssr && ctx.reactRouterConfig.prerender == null
+      };
```

Comment:
"SPA Mode" is specifically no SSR and only generating a root `index.html` - this `isSpaMode` flag is what tells the server.client not to SSR/hydrate beyond the root route.